src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -126,7 +126,7 @@ protected function getAttributes($object, $format = null, array $context)
             return $allowedAttributes;
         }
 
-        if (isset($this->attributesCache[$class])) {
+        if (isset($this->attributesCache[$class]) && !isset($context['attributes'])) {
```

Comment:
If `$context['attributes']` is defined, we will fill the ` $this->attributesCache[$class]` array with useless values.

IMHO, the correct patch is:

```php
if (!isset($context['attributes']}
     $this->extractAttributes($object, $format, $context);
}
```
before the cache layer

Reply:
Good point, updated.